Azure/azure-cli
# [ARM] `az stack`: Update to API version `2024-03-01`, add new `validate` subcommand, and updates to `create` & `delete` subcommands

**Related command**

`az stack group create`
`az stack sub create`
`az stack mg create`
`az stack group delete`
`az stack sub delete`
`az stack mg delete`
`az stack group validate`
`az stack sub validate`
`az stack mg validate`

**Description**
- Updates CLI to use the first stable GA API version `2024-03-01` for `Microsoft.Resources/deploymentStacks`.
- BREAKING CHANGE: Remove the deprecated `--delete-all`, `--delete-resources`, and `--delete-resource-groups` flags for stack create and delete commands. These are replaced by the already existing parameter `--action-on-unmanage/--aou <value>`.
- Add new `az stack <scope> validate` commands to perform preflight validation on a deployment stack deployment. This allows validating a stack's deployment before creating it or updating an existing one.
- The `az stack group create` and `az stack sub create` commands have been modified to perform validation immediately before creating or updating the stack. This is how `az deployment <scope> create` commands already behave. This does not yet apply to the `az stack mg create` command due to a backend bug with ARM and a fix that is currently awaiting deployment. This will be resolved in an upcoming bugfix release.
- Add correlation id property to stack create command output.
- Add new `--bypass-stack-out-of-sync-error` parameter to `stack create` and `stack delete` subcommands designed to unblock stacks in a stuck state due to resource sync errors.

**Testing Guide**

`az stack validate`: new command, see E2E tests for inputs
`az stack <group/sub/mg> <create/delete>`: test all variants of `--action-on-unmanage <detachAll/deleteResources/deleteAll>`, E2E tests cover this

**History Notes**


[ARM] BREAKING CHANGE: `az stack group/sub/mg create/delete`: Remove the deprecated `--delete-all`, `--delete-resources`, and `--delete-resource-groups` flags. Use the `--action-on-unmanage <value>`/`--aou <value>` parameter instead. Previously with no `--delete-*` flags specified, the default behavior was "detachAll" which can be represented with `--action-on-unmanage detachAll`
[ARM] `az stack group/sub/mg validate`: Add new `validate` command to preform preflight validation on a stack deployment.
[ARM] `az stack group/sub create`: Validation of a stack will now occur before a stack is created or updated. This is the same operation carried out by `az stack <scope> validate`. This will come to `az stack mg create` in a future bugfix release. 
[ARM] `az stack group/sub/mg create/delete`: Action on unmanage behavior for stack managed management groups can now be configured. 
[ARM] `az stack group/sub/mg create`: The correlation ID of the create operation is now returned as a property of the stack
[ARM] `az stack group/sub/mg create/delete`: Add new flag `--bypass-stack-out-of-sync-error`/`--bse` that will bypass errors related to the resource list of a stack being out of sync. 


---

This checklist is used to make sure that common guidelines for a pull request are followed.
